conan/tools/premake/premake.py

```diff
@@ -1,40 +1,96 @@
+import textwrap
+from pathlib import Path
+
+from jinja2 import Template
+
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import save
+from conan.tools.microsoft.msbuild import MSBuild
+from conan.tools.premake.toolchain import PremakeToolchain
+from conan.tools.premake.constants import CONAN_TO_PREMAKE_ARCH
+
 # Source: https://learn.microsoft.com/en-us/cpp/overview/compiler-versions?view=msvc-170
 PREMAKE_VS_VERSION = {
-    '190': '2015',
-    '191': '2017',
-    '192': '2019',
-    '193': '2022',
-    '194': '2022',  # still 2022
+    "190": "2015",
+    "191": "2017",
+    "192": "2019",
+    "193": "2022",
+    "194": "2022",  # still 2022
 }
 
-
 class Premake:
     """
     Premake cli wrapper
     """
 
+    filename = "conanfile.premake5.lua"
+
+    # Conan premake file which will preconfigure toolchain and then will call the user's premake file
+    _premake_file_template = textwrap.dedent(
+        """\
+    #!lua
+    include("{{luafile}}")
+    include("{{premake_conan_toolchain}}")
+    """
+    )
+
     def __init__(self, conanfile):
         self._conanfile = conanfile
-
-        self.action = None              # premake5 action to use (Autogenerated)
-        self.luafile = 'premake5.lua'   # Path to the root (premake5) lua file
+        # Path to the root (premake5) lua file
+        self.luafile = (Path(self._conanfile.source_folder) / "premake5.lua").as_posix()
+        print("luafile", self.luafile)
         # (key value pairs. Will translate to "--{key}={value}")
-        self.arguments = {}             # https://premake.github.io/docs/Command-Line-Arguments/
+        self.arguments = {}  # https://premake.github.io/docs/Command-Line-Arguments/
+        # self.arguments["scripts"] = Path(self._conanfile.generators_folder).as_posix()
+
+        arch = str(self._conanfile.settings.arch)
+        if arch not in CONAN_TO_PREMAKE_ARCH:
+            raise ConanInvalidConfiguration(
+                f"Premake does not support {arch} architecture."
+            )
+        self.arguments["arch"] = CONAN_TO_PREMAKE_ARCH[arch]
 
         if "msvc" in self._conanfile.settings.compiler:
-            msvc_version = PREMAKE_VS_VERSION.get(str(self._conanfile.settings.compiler.version))
-            self.action = f'vs{msvc_version}'
+            msvc_version = PREMAKE_VS_VERSION.get(
+                str(self._conanfile.settings.compiler.version)
+            )
+            self.action = f"vs{msvc_version}"
         else:
-            self.action = 'gmake2'
+            self.action = "gmake"
```

Comment:
Why? 
Seems deprecated, and the recommended one is ``gmake2``? 
https://github.com/premake/premake-core/wiki/Using-Premake/9507d6baa7b52ac97f3eaa7eddaa5f6bb984fa39

Reply:
It used to be `gmake2` the new one, but the attached documentation is already deprecated.
This is the newest documentation: https://premake.github.io/docs/Using-Premake
`gmake` is the newest one and now, the legacy generator is called `gmakelegacy`.

See `$ premake --help` output:
``` 
...
ACTIONS

 clean             Remove all binaries and generated files
 codelite          Generate CodeLite project files
 gmake             Generate GNU makefiles for POSIX, MinGW, and Cygwin
 gmakelegacy       Generate GNU makefiles for POSIX, MinGW, and Cygwin
...
```